src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -406,24 +408,58 @@ def _handleNoStatusUpdate(self, workload, request_args, dn):
-        elif workqueue_stat_validation(request_args):
-            report = self.reqmgr_db_service.updateRequestStats(workload.name(), request_args)
-            cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), request_args))
-        else:
-            msg = "There are invalid arguments for no-status update: %s" % request_args
+        report = []
+        reqArgsDiff = deepcopy(request_args)
+
+        # NOTE: Here we need to re-add `RequestName` since down bellow will be the second call of `validate_request_update_args`
+        #       which, by itself is a destructive function, which changes `request_args` with the following line:
+        #           request_name = request_args.pop("RequestName")
+        #       So if we call this function more than once inside the same request instance on every next call it will raise:
+        #           `KeyError: 'RequestName'`.
+        #       The first time it was called was during the validation of the REST call.
+        if 'RequestName' not in reqArgsDiff:
+            reqArgsDiff['RequestName'] = workload.name()
+        _, reqArgsDiff = validate_request_update_args(reqArgsDiff, self.config, self.reqmgr_db_service, None)
+
+        def _reduceReport(report):
+            if reduce(lambda x, y: x == y == 'OK' and 'OK', report, 'OK'):
+                return 'OK'
+            return False
+
+        if not reqArgsDiff:
+            cherrypy.log("Nothing to be changed at this stage")
+            return _reduceReport(report)
+
+        reqArgsNothandled = []
+        for reqArg in reqArgsDiff.copy():
+            if reqArg == 'RequestPriority':
+                validate_request_priority(reqArgsDiff)
+                # must update three places: GQ elements, workload_cache and workload spec
+                self.gq_service.updatePriority(workload.name(), reqArgsDiff['RequestPriority'])
+                report.append(self.reqmgr_db_service.updateRequestProperty(workload.name(), reqArgsDiff, dn))
+                workload.setPriority(reqArgsDiff['RequestPriority'])
+                workload.saveCouchUrl(workload.specUrl())
+                cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), reqArgsDiff['RequestPriority']))
+            elif workqueue_stat_validation(reqArgsDiff):
+                report.append(self.reqmgr_db_service.updateRequestStats(workload.name(), reqArgsDiff))
+                cherrypy.log('Updated workqueue statistics of "{}", with:  {}'.format(workload.name(), reqArgsDiff))
+            elif reqArg == "SiteWhitelist":
+                report.append(self.reqmgr_db_service.updateRequestProperty(workload.name(), reqArgsDiff, dn))
+                workload.setSiteWhitelist(reqArgsDiff["SiteWhitelist"])
+                cherrypy.log('Updated SiteWhitelist of "{}", with:  {}'.format(workload.name(), reqArgsDiff['SiteWhitelist']))
+            elif reqArg == "SiteBlacklist":
```

Comment:
Alan, you are inside a `for` cycle here. Every check you do at  this stage is always a single possible match during one iteration of the cycle. So it actually spares you extra/redundant steps, because the moment the parameter matches check we take the relevant action and we get out of the `elif` sequence to continue with the next iteration of checks  for the next request parameter from the `for` cycle. The other possibility is to use multiple independent `if` statements, which would have exactly the same effect, but at the cost of iterating through all checks every time, regardless if we have found a match or not.

Reply:
Yes, I overlooked the fact that it was inside a for loop.

I see you have separated the call to `self.reqmgr_db_service`, which can be either for `updateRequestStats` or for `updateRequestProperty`. Those are exclusive, so there is no need to deal with `report` as a list and provide the extra complexity with the reduceReport function.

Reply:
> Those are exclusive, so there is no need to deal with report as a list 

I did not get this comment. We do need to accumulate the reports in a list exactly because of the calls to different methods from the `reqmgr_db_service`. All of them use the same upstream mechanisms and functionalities  from couchdb, so the set of report values from both types of calls are similar, ergo applying identical criteria for checking the reports  from all of them is feasible. And the fact that we can get a request for changing multiple request parameters at once, leads to the need  of those reports being accumulated in a single list and later crosschecked if any of the couchdb calls failed. So to me this is the logical and sensible approach.

Reply:
Todor, I followed your code once again in a clean view (without GH diff):
https://github.com/dmwm/WMCore/blob/0da38322dfb816bd00b0df079cf24d58a9060966/src/python/WMCore/ReqMgr/Service/Request.py

and I do not see multiple requests being consumed in any moment. It is always returned as a single call report! So please remove the call to reduceReport() in (currently) L455.

Reply:
well ... before we start this whole process of changing everything back and forth multiple  times ... the report results were accumulated while iterating through the request parameters. They have never been intended to be accumulated through iteration over multiple request - because it simply never happens.  But none of this matters any more, because as I sent you in a private chat this whole `for` cycle here I removed in my next PR, where I make this whole set of calls to the `workload` and `global workque` a single action. Which is actually the main challenge of the next issue to fight in this line of coding. The whole thing is reduced to just two lines like this: 
https://github.com/dmwm/WMCore/pull/12096/files#diff-120ee6838284a3d1c1799f511da7f147179d0a955f87d0da6fc8b58a8b66c794R428-R462
```
        # reqArgsNothandled = []
        # for reqArg in reqArgsDiff.copy():
        #     if reqArg == 'RequestPriority':
        #         validate_request_priority(reqArgsDiff)
        #         # must update three places: GQ elements, workload_cache and workload spec
        #         self.gq_service.updatePriority(workload.name(), reqArgsDiff['RequestPriority'])
        #         workload.setPriority(reqArgsDiff['RequestPriority'])
        #         cherrypy.log('Updated priority of "{}" to: {}'.format(workload.name(), reqArgsDiff['RequestPriority']))
        #     elif reqArg == "SiteWhitelist":
        #         workload.setSiteWhitelist(reqArgsDiff["SiteWhitelist"])
        #         cherrypy.log('Updated SiteWhitelist of "{}", with:  {}'.format(workload.name(), reqArgsDiff['SiteWhitelist']))
        #     elif reqArg == "SiteBlacklist":
        #         workload.setSiteBlacklist(reqArgsDiff["SiteBlacklist"])
        #         cherrypy.log('Updated SiteBlacklist of "{}", with:  {}'.format(workload.name(), reqArgsDiff['SiteBlacklist']))
        #     else:
        #         reqArgsNothandled.append(reqArg)
        #         cherrypy.log("Uunhandled argument for no-status update: %s" % reqArg)


        # TODO:  map setters to key names:
        # Creating a setter method map - directly in the workload object.

        # Update all workload parameters based on the full reqArgsDiff dictionary
        workload.updateWorkloadArgs(reqArgsDiff)

        # Commit the changes of the current workload object to the database:
        workload.saveCouchUrl(workload.specUrl())

        # Commit all Global WorkQueue changes per workflow in a single go:
        # self.gq_service.updateElementsByWorkflow(workload.name(), reqArgsDiff)

        # if reqArgsNothandled:
        #     msg = "There were unhandled arguments left for no-status update: %s" % reqArgsNothandled
        #     raise InvalidSpecParameterValue(msg)
```

P.S. the commented lines I left just for now to point the code evolution. they will all be removed.

Reply:
And I also rearranged the code in this PR such that nothing is persisted in the Database in case of unsuported arguments sent by the user, so that we can merge this and move on.